#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: The patch below enhances the FBIOGET_FSCREENINFO ioctl.
## DP: Description: It is used by the lcd_hack tool that Ben H wrote. Before
## DP: Description: this fix accel is set to 0 and lcd_hack aborts. With it,
## DP: Description: the tool works again.
## DP: Patch author: Martin Habets <errandir_news@mph.eclipse.co.uk>
## DP: Upstream status: approved by benh, no idea about upstream status

. $(dirname $0)/DPATCH

@DPATCH@
--- kernel-source-2.6.11-2.6.11-orig/drivers/video/aty/radeon_base.c	2005-03-02 08:37:54.000000000 +0100
+++ kernel-source-2.6.11-2.6.11/drivers/video/aty/radeon_base.c	2005-03-05 10:37:19.668812008 +0100
@@ -1832,8 +1832,12 @@
 
 	fb_alloc_cmap(&info->cmap, 256, 0);
 
-	if (noaccel)
+	if (noaccel) {
 		info->flags |= FBINFO_HWACCEL_DISABLED;
+		info->fix.accel = FB_ACCEL_NONE;
+	} else {
+		info->fix.accel = FB_ACCEL_ATI_RADEON;
+	}
 
         return 0;
 }
